### PR TITLE
Disallow horizontal scrolling in health checkup

### LINKED ISSUE #30: [BUG] "Your Health Checkup" doesn't have a bottom bar when contributing data
If you select "Contribute my data", the next screen is missing the contents of the footer (though the footer itself is still there:

Screenshot_20200322-151131

### LINKED ISSUE #37: [BUG] Using a scrollable page view in the checkup form is unexpected UX
**Describe the bug**
Either emphasize the scrollable nature with page dot indicators (but uncommon UI for pages with user entry) or make it advance/reversible via buttons only.

### PR DESCRIPTION
**Pull request by goderbauer**
Changes in this PR:
* The health checkup no longer scrolls in horizontal direction (users will have to use the continue button)
* The progress bar always appears even if location setup fails.
* The progress bar animated in.
* Small a11y fix to progress bar.

Fixes https://github.com/coronavirus-diary/coronavirus-diary/issues/37
Fixes https://github.com/coronavirus-diary/coronavirus-diary/issues/30

### CODE REVIEW

**Review by dnfield — COMMENTED** [review]
Change looks good, can we add a test for this?  E.g. that the screen can't be swiped away?

File: lib/src/ui/screens/checkup/checkup_loaded_body.dart [hunk 1]
```diff
@@ -49,17 +49,17 @@ class _CheckupLoadedBodyState extends State<CheckupLoadedBody> {
...
+    });
     // Origin-specific actions
-    if (currentIndex == 0) {
+    if (oldIndex == 0) {
       if (checkupState.checkup.dataContributionPreference == true) {
         await _saveCurrentLocation(checkupState);
       }
     }
-
-    setState(() {
```
Comment (xster) [inline lib/src/ui/screens/checkup/checkup_loaded_body.dart:59]: why move this? This is assuming that the save location above will never be changed to throw and not change page right?

**Review by xster — DISMISSED** [review]
Thanks for fixing this

File: lib/src/ui/screens/checkup/checkup_loaded_body.dart (hunk 1 above)
Comment (goderbauer) [reply lib/src/ui/screens/checkup/checkup_loaded_body.dart:59]: We should always advance the state even if the asynchronous call above takes longer.

Comment (xster) [reply lib/src/ui/screens/checkup/checkup_loaded_body.dart:59]: sounds reasonable

**Comment by goderbauer** [comment]
@dnfield I am trying to figure out how to write a test for this, but so far I am unable to get the `CheckupBloc` into the right state for the test.

**Comment by goderbauer** [comment]
@dnfield tests added with special thanks to @jonahwilliams for the guidance in writing tests for this.

**Review by dnfield — DISMISSED** [review]
LGTM

**Review by ferndot — DISMISSED** [review]
Looks good to me as well 👍

**Review by dnfield — DISMISSED** [review]

**Review by dnfield — APPROVED** [review]

**Pull request merged**